### ISSUE TITLE
Inductive_mode model returns all NA prediction when using ZINC split data

### ISSUE THREAD

**Issue by BIOBRICK** [HamidHadipour/GraphBAN#8]
Thank you for such a good work. However, I am feeling confused about those "trained models" in git repo and zenodo.
 I used following script and get only NA prediction on zinc_split_1.csv you provided:
`nohup python predict.py --folder_path ../inductive_mode/trained_models/bindingdb/inductive/seed12/result/ --test_path ./zinc_data/split_zinc_1.csv --save_dir ../../data/Predictions/ProteomeScreen.csv &`

the log says:

 99%|█████████▉| 9892/10000 [00:30<00:00, 322.59it/s]
 99%|█████████▉| 9925/10000 [00:30<00:00, 322.64it/s]
100%|█████████▉| 9958/10000 [00:30<00:00, 321.67it/s]
100%|█████████▉| 9991/10000 [00:31<00:00, 321.76it/s]
                       
/data/software/miniconda3/envs/graphban/lib/python3.11/site-packages/torch/nn/utils/weight_norm.py:28: UserWarning: torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.
  warnings.warn("torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.")
0
   row_average
0          NaN
1          NaN
2          NaN
3          NaN
4          NaN
                                              SMILES  ... predicted_value
0          CC(C)(C)c1ccc2occ(CC(=O)Nc3ccccc3F)c2c1\n  ...             NaN
1     C[C@@H]1CC(Nc2cncc(-c3nncn3C)c2)C[C@@H](C)C1\n  ...             NaN
2  N#Cc1ccc(-c2ccc(O[C@@H](C(=O)N3CCCC3)c3ccccc3)...  ...             NaN
3  CCOC(=O)[C@@H]1CCCN(C(=O)c2nc(-c3ccc(C)cc3)n3c...  ...             NaN
4  N#CC1=C(SCC(=O)Nc2cccc(Cl)c2)N=C([O-])[C@H](C#...  ...             NaN

I don't understand **what's the difference between the model weight you provide in git repo in inductive/transductive folder and those bindingdb_trained_models.zip in zenodo.** And I don't know why I get all NA here. Even the model is not that good. I should get some value.

Can you identify the problem? Or **can I only use model weights in zenodo when I need to use my own screen data**?

Besides, I'm wondering **whether your model can be used on reverse DTI situation or other species, like E.coli protein and SMILES?** (Just think)

Any help will be very useful, hope this message can reach you asap, thank you very much!!!

**Comment by BIOBRICK** [comment]
OK I know now.
I should download the weights from zenodo and use it to predict my data. And the problem goes away!

However, I still want to know what's the difference between all those weights in, such as result_bindingdb_zincxxx? I see multiple weight folders and is there any priciple to guide us to chose any one of them?
Any help will be grateful and thanks a lot for your work again!

**Comment by HamidHadipour** [comment]
Thanks for your interest in this project!
The number of models you mentioned in folders like result_bindingdb_zincxxx are models saved for each of 50 epochs during training. The point is that in the case study, we do not have access to the true labels, and as a result, we do not know about the performance of the model in each epoch, so we gathered all and used all to get the predictions, and after all, we got the average of all 50 models. You can try with a lower number of epochs, like 20, for your initial tests.

**Comment by BIOBRICK** [comment]
> Thanks for your interest in this project! The number of models you mentioned in folders like result_bindingdb_zincxxx are models saved for each of 50 epochs during training. The point is that in the case study, we do not have access to the true labels, and as a result, we do not know about the performance of the model in each epoch, so we gathered all and used all to get the predictions, and after all, we got the average of all 50 models. You can try with a lower number of epochs, like 20, for your initial tests.

My apologies, I didn't make it clear. I fully understand what you mean. However,  I mean:
When I download the trained model weight from zenodo(https://zenodo.org/records/14813233), I find a lot of dirs in same trained folder. like below:

├── result_bindingdb12_zinc21
├── result_bindingdb12_zinc210
├── result_bindingdb12_zinc211
├── result_bindingdb12_zinc212
├── result_bindingdb12_zinc213
├── result_bindingdb12_zinc214
├── result_bindingdb12_zinc215
├── result_bindingdb12_zinc216
├── result_bindingdb12_zinc217
├── result_bindingdb12_zinc218
├── result_bindingdb12_zinc219
├── result_bindingdb12_zinc22
├── result_bindingdb12_zinc220
├── result_bindingdb12_zinc221
├── result_bindingdb12_zinc222
├── result_bindingdb12_zinc223
├── result_bindingdb12_zinc224
├── result_bindingdb12_zinc225
├── result_bindingdb12_zinc23
├── result_bindingdb12_zinc24
├── result_bindingdb12_zinc25
├── result_bindingdb12_zinc26
├── result_bindingdb12_zinc27
├── result_bindingdb12_zinc28
└── result_bindingdb12_zinc29

And in every result_bindingdb12_zincXX fold, there are lots of pth weight files like:
├── result_bindingdb12_zinc29
│   ├── model_epoch_10.pth
│   ├── model_epoch_11.pth
│   ├── model_epoch_12.pth
│   ├── model_epoch_13.pth
│   ├── model_epoch_14.pth
│   ├── model_epoch_15.pth
│   ├── model_epoch_16.pth
│   ├── model_epoch_17.pth
│   ├── model_epoch_18.pth
│   ├── model_epoch_19.pth
│   ├── model_epoch_1.pth
│   ├── model_epoch_20.pth
│   ├── model_epoch_21.pth
│   ├── model_epoch_22.pth
│   ├── model_epoch_23.pth
│   ├── model_epoch_24.pth
│   ├── model_epoch_25.pth
│   ├── model_epoch_26.pth
│   ├── model_epoch_27.pth
│   ├── model_epoch_28.pth
│   ├── model_epoch_29.pth
│   ├── model_epoch_2.pth
│   ├── model_epoch_30.pth
│   ├── model_epoch_31.pth
│   ├── model_epoch_32.pth
│   ├── model_epoch_33.pth
│   ├── model_epoch_34.pth
│   ├── model_epoch_35.pth
│   ├── model_epoch_36.pth
│   ├── model_epoch_37.pth
│   ├── model_epoch_38.pth
│   ├── model_epoch_39.pth
│   ├── model_epoch_3.pth
│   ├── model_epoch_40.pth
│   ├── model_epoch_41.pth
│   ├── model_epoch_42.pth
│   ├── model_epoch_43.pth
│   ├── model_epoch_44.pth
│   ├── model_epoch_45.pth
│   ├── model_epoch_46.pth
│   ├── model_epoch_47.pth
│   ├── model_epoch_48.pth
│   ├── model_epoch_49.pth
│   ├── model_epoch_4.pth
│   ├── model_epoch_50.pth
│   ├── model_epoch_5.pth
│   ├── model_epoch_6.pth
│   ├── model_epoch_7.pth
│   ├── model_epoch_8.pth
│   ├── model_epoch_9.pth
│   ├── result_metrics.pt
│   ├── test_markdowntable.txt
│   ├── test_zinc9_preds.csv
│   ├── train_markdowntable.txt
│   └── valid_markdowntable.txt

I'm asuming the model_epoch_x.pth is the weight file you menthioned. **But I want to know, which "folder" should I use?**

Thank you for your patience and hope you well!
